### ISSUE TITLE
Fix ignoring of files when code and codeFilename properties are used

### ISSUE THREAD

**Issue by shinnn** [stylelint/stylelint#3590]
> Clearly describe the bug

Even if a file path passed to [`codeFilename`](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/node-api.md#codefilename) option should be ignored by `.stylelintignore`, it's still validated as if it's not ignored.

> Which rule, if any, is the bug related to?

Rules are not related to this issue.

> What CSS is needed to reproduce the bug?

Any invalid CSS, for example `}`.

> What stylelint configuration is needed to reproduce the bug?

Any configurations, even if you enable no rules in your `config`.

> Which version of stylelint are you using?

`9.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

In the current working directory, you have the following files

**.stylelintignore**

```
style.css
```

**style.css**

```css
}
```

As you can see, `style.css` is not a valid CSS.

**files.js**

```js
'use strict';

const {inspect} = require('util');
const {join} = require('path');
const {lint} = require('stylelint');

(async () => {
  const result = await lint({
    files: join(__dirname, 'style.css'),
    config: {rules: {}}
  });

  console.log(inspect(result, {colors: true, depth: Infinity}));
})();
```

**code.js**

```js
'use strict';

const {inspect} = require('util');
const {join} = require('path');
const {lint} = require('stylelint');

(async () => {
  const result = await lint({
    code: '}',
    codeFilename: join(__dirname, 'style.css'),
    config: {rules: {}}
  });

  console.log(inspect(result, {colors: true, depth: Infinity}));
})();
```

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

When you run `node files.js`, it prints

```
{ errored: false, output: '[]', results: [] }
```

It's OK. `style.css` is successfully ignored by `.stylelintignore`.

Then, I expect `node code.js` to results in exactly the same output, as `code.js` and `file.js` do essentially the same thing.

> What actually happened (e.g. what warnings or errors did you get)?

`node code.js` prints:

```
{ errored: true,
  output: '',
  results:
   [ { source: '/Users/shinnn/example/style.css',
       deprecations: [],
       invalidOptionWarnings: [],
       parseErrors: [],
       errored: true,
       warnings:
        [ { line: 1,
            column: 1,
            rule: 'CssSyntaxError',
            severity: 'error',
            text: 'Unexpected } (CssSyntaxError)' } ] } ] }
```

This is unexpected result. `style.css` should be ignored by `.stylelintignore`, but it's actually validated and emits `CssSyntaxError`.

**Comment by shinnn** [comment]
Even if you pass `ignorePath: join(__dirname, '.stylelintignore')` to the options object explicitly, this bug still happens.

**Comment by thibaudcolas** [comment]
From what I understand this is the same issue as #2833. The reason why @shinnn's results above are different form mine is that I was only looking at the `output` part of the linting results… which is where this error lies 😄.
